vignettes/quality_control_workflow.Rmd

```diff
@@ -0,0 +1,304 @@
+
+This document will give you an overview of the QC functions included in **protti** and will show you how these can be applied to your data. The examples in this file are run on synthetic data, which is created with the function `create_synthetic_data()`.
+
+The functions in **protti** are tailored towards the analysis of DIA data, based on the output of the search engine [Spectronaut™](https://biognosys.com/shop/spectronaut). However, if you have any other data, such as DDA data that was searched with a different search engine, you can still apply our functions. Just make sure that the formatting of your data frame is the same as in our example data (peptides, precursors or proteins as rows, samples as rows and a corresponding intensity column). If you are unsure about how your input data is supposed to look like, please use the `create_synthetic_data()` function and compare this to your data. 
+
+The input data should look something like that: 
+
+| Sample  |  Replicate |  Peptide  |  Condition | log2(intensity) |
+|:-------:|:----------:|:---------:|:----------:|:----------------:
+| sample1 |      1     |  PEPTIDER |   treated  |     14          |
+| sample1 |      1     |  PEPTI    |   treated  |     16          |
+| sample1 |      1     |  PEPTIDE  |   treated  |     17          |
+| sample2 |      1     |  PEPTIDER |  untreated |     15          |
+| sample2 |      1     |  PEPTI    |  untreated |     18          |
+| sample2 |      1     |  PEPTIDE  |  untreated |     12          |
+
+## Getting started
+
+Before we can start analysing our data, we need to load the **protti** package. This is done by using the base R function `library()`. In addition, we are also loading the packages `magrittr` and `dplyr`.
+
+```{r setup, message = FALSE}
+library(protti)
+library(magrittr)
+library(dplyr)
+
+```
+
+After having loaded our required packages, we can create a synthetic dataset, as it should be obtained though a treatment experiment with for example a protein, metabolite or small-molecule. 
+You can skip this step if you have your own data ready.
+
+We are creating a random dataset with 100 different proteins, out of which 5 % are significantly changing upon treatment. The dataset includes 3 replicates for 2 different conditions (treated and untreated).
+
+The `create_synthetic_data()` function is modeled after real LiP-MS data and its commonly observed data distributions.
+
+Please note that generally, quality control should be conducted on raw unfiltered data (the direct output of your search engine of choice).
+
+```{r create_synthetic_data}
+set.seed(123) # by setting the seed we are making sure that the random object generation can be reproduced
+
+data <- create_synthetic_data(
+  n_proteins = 100,
+  frac_change = 0.05,
+  n_replicates = 3,
+  n_conditions = 2,
+  method = "random_effect",
+  additional_metadata = TRUE
+)
+
+```
+
+## Quality control
+
```

Comment:
The way I usually set up the quality control is to first look at qualitative quality (everything that does not involve intensities). Then to look at the quantitative quality. 
For the quantitative I would potentially start first with the distribution of data e.g. qc_log2_intensity_distribution. Just to see that everything looks normal and not with the CVs.